src/main/java/org/fenixedu/academic/ui/spring/AcademicBaseController.java

```diff
@@ -0,0 +1,39 @@
+package org.fenixedu.academic.ui.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+
+public class AcademicBaseController {
+    //The HTTP Request that can be used internally in the controller
```

Comment:
As you're only using this in one controller, just put this implementation there, and it will eventually be pulled up in the future, if necessary.

[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)